sphinx_click/ext.py

```diff
@@ -428,74 +497,126 @@ def _generate_nodes(
             empty
         :param semantic_group: Display command as title and description for
             `click.CommandCollection`.
+        :param hide_header: Hide the title and summary.
         :returns: A list of nested docutil nodes
         """
         ctx = click.Context(command, info_name=name, parent=parent)
-
+        
         if command.hidden:
             return []
 
-        # Title
-
-        section = nodes.section(
-            '',
-            nodes.title(text=name),
-            ids=[nodes.make_id(ctx.command_path)],
-            names=[nodes.fully_normalize_name(ctx.command_path)],
-        )
-
         # Summary
         source_name = ctx.command_path
         result = statemachine.ViewList()
 
+        lines = iter(())
+        hide_current_header = hide_header
+        if nested == NESTED_COMPLETE:
+            lines = itertools.chain(lines, _format_summary(ctx, commands, hide_header=hide_header))
+            nested = NESTED_FULL
+            hide_current_header = True
+        
         if semantic_group:
-            lines = _format_description(ctx)
+            lines = itertools.chain(lines, _format_description(ctx))
         else:
-            lines = _format_command(ctx, nested, commands)
+            lines = itertools.chain(lines, _format_command(ctx, nested, commands, hide_header=hide_current_header))
 
         for line in lines:
             LOG.debug(line)
             result.append(line, source_name)
 
-        sphinx_nodes.nested_parse_with_titles(self.state, result, section)
-
         # Subcommands
 
+        subcommand_nodes = []
         if nested == NESTED_FULL:
             if isinstance(command, click.CommandCollection):
                 for source in command.sources:
-                    section.extend(
+                    subcommand_nodes.extend(
                         self._generate_nodes(
                             source.name,
                             source,
                             parent=ctx,
                             nested=nested,
                             semantic_group=True,
+                            hide_header=False,  # Hiding the header should not propagate to children
                         )
                     )
             else:
                 commands = _filter_commands(ctx, commands)
-                for command in commands:
+                for subcommand in commands:
```

Comment:
Although we use the term "subcommands" above I'm not sure it's correct. For example, if listing commands in a group, these are really commands, not subcommands. Can we keep the existing name?

Reply:
If you'd like to use another name, that's fine, but `command` is the name of the parameter you get in the `_generate_nodes()` function, of which I added a use at the end of the function to invoke the post-processor.
This loop was overriding the contents of said parameter, so that instead of the post-processor being invoked with the original command, it was instead invoked with the last value that was put in that variable during this loop.
So we can either:
1. Rename this loop variable.
2. Rename the original parameter.
3. Make a copy of the original parameter in another variable before performing the loop.
Personally I figured renaming the loop variable makes the most sense so I went with that.
Please tell me which option you prefer and I'll change it accordingly.

Reply:
Fair point. Let's stick with `subcommand` and just add a comment indicating that these are really commands if `command` is a group?